prow/jobs.md

```diff
@@ -31,14 +31,32 @@ that you can define pods in yaml.  Please see kubernetes documentation
 for help here, for example the [Pod overview] and [PodSpec api
 reference].
 
-Periodic config looks like so (see [GoDocs](https://pkg.go.dev/k8s.io/test-infra/prow/config#Periodic) for complete config):
+### Types of jobs
+
+There are 3 basic types of jobs:
+
+1. periodics
+2. presubmits
+3. postsubmits
+
+They are distinguished by **when** they run. **Periodics** run at a regular
```

Comment:
Another important part of this concept is that presubmits and postsubmits not only trigger **when** refs are pushed, but they are actually intrinsically associated with a specific git ref (either a specific ref of a specific PR or a specific ref that has been pushed to a branch). By this I mean that those git refs are dynamically specified based on the triggering event and that Prow knows to handle the job appropriately for the given refs (e.g. running presubmits will be aborted when new commits are pushed to the PR, Tide will retest presubmits if and only if needed, Crier will report presubmits back to the correct PRs based on the refs, etc.)

Periodics on the other hand are not intrinsically associated with any git refs. Prow can check out source for any ProwJob including periodics using `extra_refs`, but the dynamically supplied refs are only applicable to presubmits and postsubmits.

Reply:
I think the ref association concept is too much for users to handle at this point in this doc. It's interesting info, and if it's not recorded anywhere else we should record it maybe near the end of this doc for now. WDYT?

Reply:
I agree that what I shared is too much info, definitely would be overwhelming for a reader. But I do think there is an important piece in there that we should highlight. Maybe just explaining that presubmits and postsubmits automatically check out the git ref that triggered them? If you think that is already clear from the doc (or obvious based on name/context) feel free to ignore this comment, I just want to make sure we get the point across that presubmits and postsubmits have a dynamic git ref and periodics don't.